google/cloud/spanner/database_admin_client.h

```diff
@@ -392,12 +394,37 @@ class DatabaseAdminClient {
    * @snippet samples.cc create-backup
    */
   future<StatusOr<google::spanner::admin::database::v1::Backup>> CreateBackup(
-      Database db, std::string backup_id,
-      std::chrono::system_clock::time_point expire_time,
-      absl::optional<std::chrono::system_clock::time_point> version_time =
-          absl::nullopt,
+      Database db, std::string backup_id, Timestamp expire_time,
+      absl::optional<Timestamp> version_time = absl::nullopt,
       absl::optional<KmsKeyName> encryption_key = absl::nullopt);
 
+  /**
+   * Creates a new Cloud Spanner backup for the given database.
+   *
+   * @par Idempotency
+   * This is not an idempotent operation. Transient failures are not retried.
+   *
+   * This function creates a database backup for the given Google Cloud Spanner
+   * database.
+   *
+   * Note that the @p backup_id must be unique within the same instance, it must
+   * be between 2 and 60 characters long, it must start with a lowercase letter
+   * (`[a-z]`), it must end with a lowercase letter or a number (`[a-z0-9]`) and
+   * any characters between the beginning and ending characters must be lower
+   * case letters, numbers, underscore (`_`) or dashes (`-`), that is, they must
+   * belong to the `[a-z0-9_-]` character set.
+   *
+   * The @p expire_time must be at least 6 hours and at most 366 days from the
+   * time the `CreateBackup()` request is processed.
+   *
+   * @return A `google::cloud::future` that becomes satisfied when the operation
+   *   completes on the service. Note that this can take minutes in some cases.
+   */
+  ABSL_DEPRECATED("Use the Timestamp overload instead")
```

Comment:
My understanding of this attribute is that it will NOT produce a warning by default under clang for gcc, but it will cause a warning from clang-tidy? I believe there's also a compiler flag that if enabled will cause this to emit a warning. So the end result is that for most of our callers will this be silent and not break their builds. However, for customers using clang-tidy they'll start seeing this warning. Additionally, for any customers compiling with the "show deprecation warnings" flag enabled, their build might break (if using -Werror).

Does this sound right to you?

Reply:
I do see warnings like ...
```
google/cloud/spanner/integration_tests/backup_integration_test.cc:146:30: warning: 'CreateBackup' is deprecated: Use the Timestamp overload instead [-Wdeprecated-declarations]
```
from clang (but I don't know how `deprecated-declarations` was enabled, in this case in ubsan, or why it might not be in other builds).

So, I guess I'll say that sounds right, and working as advertised.

Reply:
I couldn't find those warnings in the build logs. Which build? Do you have a link?

If those show up all the time, might it break users who compile with `-Werror`?

To be clear, my concern is that if adding a formal deprecation tag warning here may break users, we may want to consider whether we actually want to use that tag, or if we just want to document that an overload/field is going away. I don't think we've used `ABSL_DEPRECATED()` yet, and I'm afraid I don't yet really understand its implications. If you do, please feel free to just enlighten me so that we can feel confident using it now and going forward.